### ISSUE TITLE
Multiple deprecation warnings

### ISSUE THREAD

**Issue by elchininet** [wessberg/rollup-plugin-ts#198]
- Version: 3.0.2
- Rollup Version: 2.79.0
- Operating System and version (if applicable): MacOS 13.0.1 
- Node Version (if applicable): 16.10.0
- Does it work with `tsc` (if applicable): yes

### Reproduction

1. Clone this repo: https://github.com/elchininet/isometric
2. Update `TypeScript` from version `4.8.3` to version `4.9.4`
3. Multiple deprecation warnings are thrown:

```
DeprecationWarning: 'updateEnumDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'isTypeAssertion' has been deprecated since v4.0.0. Use `isTypeAssertionExpression` instead.
DeprecationWarning: 'createEnumDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'updateExportDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'updateModuleDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'updateTypeAliasDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createTypeAliasDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createParameterDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createInterfaceDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createModuleDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createExportDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'updateInterfaceDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createImportDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createIndexSignature' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'updateClassDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createConstructorDeclaration' has been deprecated since v4.8.0. Decorators are no longer supported for this function. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createPropertyDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createGetAccessorDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createSetAccessorDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createClassDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
DeprecationWarning: 'createMethodDeclaration' has been deprecated since v4.8.0. Decorators have been combined with modifiers. Callers should switch to an overload that does not accept a 'decorators' parameter.
```

### Expected Behavior

Should not warn about deprecated decorator parameter for multiple methods

### Actual Behavior

It warns about deprecated decorator parameter for multiple methods

### Note

Decorators are either not supported anymore or combined with modifiers for multiple methods. [This pull request](https://github.com/horiuchi/dtsgenerator/pull/547/files) in [the dtsgenerator repo](https://github.com/horiuchi/dtsgenerator) solves a similar issue removing the decorator parameter from `createModuleDeclaration`, `createInterfaceDeclaration`, `createTypeAliasDeclaration`, `createIndexSignature`, `createParameterDeclaration`, and `createIdentifier` methods.

**Comment by dfreeman** [comment]
We're also running into this in a project where we're trying to upgrade TypeScript. I know the TS team recently updated a bunch of factory signatures as part of preparing support for the stage 3 version of the decorators spec

**Comment by wessberg** [comment]
Hey there. This has been fixed as part of [v3.10.0](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v3.1.0)
